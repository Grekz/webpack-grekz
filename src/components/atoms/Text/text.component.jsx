// @flow
import * as React from 'react'

type Props = {
  As: string,
  children?: React.Node
}
const Text = ({ As = 'span', ...props }: Props) => <As>{props.children}</As>
export default Text
