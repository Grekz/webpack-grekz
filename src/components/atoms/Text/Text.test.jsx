import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Text from './text.component'

configure({ adapter: new Adapter() })

describe('<Text /> rendering', () => {
  it('should render Text', () => {
    const wrapper = mount(<Text />)
    expect(wrapper).toHaveLength(1)
  })
  it('should contain 1 p element', () => {
    const wrapper = mount(<Text />)
    expect(wrapper.find('span')).toHaveLength(1)
  })
  it('mathces the snapshot', () => {
    const tree = mount(<Text />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
