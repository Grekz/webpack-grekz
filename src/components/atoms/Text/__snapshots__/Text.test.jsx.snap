// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> rendering mathces the snapshot 1`] = `
<Text>
  <span />
</Text>
`;
